File: python/day-1b.py
def main():
    with open('input/day-1.txt', 'r') as f:
        elves = [sum(map(int, seq.split('\n')))
                 for seq in f.read().split('\n\n')]
        elves.sort(reverse=True)
        return sum(elves[:3])


print(main())

File: rust/src/bin/day-1b.rs
fn main() -> std::io::Result<()> {
    let mut lines: Vec<u32> = include_str!("../../../input/day-1.txt")
        .split("\n\n")
        .map(|x| x.split('\n').map(|x| x.parse::<u32>().unwrap()).sum())
        .collect();

    lines.sort();

    let total: u32 = (&lines[(lines.len() - 3)..lines.len()]).into_iter().sum();

    println!("{}", total);
    Ok(())
}
